modules/sdk-coin-ada/src/ada.ts

```diff
@@ -27,42 +74,166 @@ export class Ada extends BaseCoin {
   }
 
   public getChain(): string {
-    return 'ada';
+    return this._staticsCoin.name;
   }
 
-  public getFamily(): string {
-    return 'ada';
+  public getFamily(): CoinFamily {
+    return this._staticsCoin.family;
   }
 
   public getFullName(): string {
-    return 'Ada';
+    return this._staticsCoin.fullName;
   }
 
-  verifyTransaction(params: VerifyTransactionOptions): Promise<boolean> {
-    throw new Error('Method not implemented.');
+  getBaseChain(): string {
+    return this.getChain();
+  }
+
+  async verifyTransaction(params: VerifyTransactionOptions): Promise<boolean> {
+    let totalAmount = new BigNumber(0);
+    const coinConfig = coins.get(this.getChain());
+    const { txPrebuild: txPrebuild, txParams: txParams } = params;
+    const transaction = new Transaction(coinConfig);
+    const rawTx = txPrebuild.txHex;
+    if (!rawTx) {
+      throw new Error('missing required tx prebuild property txHex');
+    }
+
+    transaction.fromRawTransaction(rawTx);
+    const explainedTx = transaction.explainTransaction();
+
+    if (txParams.recipients !== undefined) {
+      const filteredRecipients = txParams.recipients?.map((recipient) => _.pick(recipient, ['address', 'amount']));
+      const filteredOutputs = explainedTx.outputs.map((output) => _.pick(output, ['address', 'amount']));
+
+      if (!_.isEqual(filteredOutputs, filteredRecipients)) {
+        throw new Error('Tx outputs does not match with expected txParams recipients');
+      }
+      for (const recipients of txParams.recipients) {
+        totalAmount = totalAmount.plus(recipients.amount);
+      }
+      if (!totalAmount.isEqualTo(explainedTx.outputAmount)) {
+        throw new Error('Tx total amount does not match with expected total amount field');
+      }
+    }
+    return true;
   }
 
   isWalletAddress(params: VerifyAddressOptions): boolean {
     throw new Error('Method not implemented.');
   }
 
-  parseTransaction(params: ParseTransactionOptions): Promise<ParsedTransaction> {
-    throw new Error('Method not implemented.');
+  /** @inheritDoc */
+  async signMessage(key: KeyPair, message: string | Buffer): Promise<Buffer> {
+    const adaKeypair = new AdaKeyPair({ prv: key.prv });
+    if (Buffer.isBuffer(message)) {
+      message = base58.encode(message);
```

Comment:
why base58 encode message? The signing is over buffer, and if message is string, it is hex encoded.